src/optimizer/query_rewrite.c

```diff
@@ -7628,6 +7628,47 @@ qo_rewrite_innerjoin (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *c
   return node;
 }
 
+/*
+ * qo_add_keylimit_clause () - Add limit clause to subquery exists
+ *   return: void
+ *   parser(in):
+ *   node(in): QUERY node
+ */
+void
+qo_add_limit_clause (PARSER_CONTEXT * parser, PT_NODE * node)
+{
+  if (node->info.query.limit != NULL || node->info.query.q.select.group_by != NULL || node->node_type != PT_SELECT)
+    {
+      return;
+    }
+
+  PT_NODE *ins_num = parser_new_node (parser, PT_VALUE);
+  ins_num->type_enum = PT_TYPE_INTEGER;
+  ins_num->info.value.data_value.i = 1;
+
+  node->info.query.limit = ins_num;
+  node->info.query.limit->next = NULL;
+  node->info.query.flag.rewrite_limit = 0;
+
+  PT_NODE *ins_num_pred = parser_new_node (parser, PT_EXPR);
+  ins_num_pred->type_enum = PT_TYPE_LOGICAL;
+  ins_num_pred->info.expr.op = PT_LE;
+
+  PT_NODE *ins_num_pred_arg1 = parser_new_node (parser, PT_EXPR);
+  ins_num_pred_arg1->type_enum = PT_TYPE_BIGINT;
+  ins_num_pred_arg1->info.expr.op = PT_INST_NUM;
+  PT_EXPR_INFO_SET_FLAG (ins_num_pred_arg1, PT_EXPR_INFO_INSTNUM_C);
+
+  PT_NODE *ins_num_pred_arg2 = parser_new_node (parser, PT_VALUE);
+  ins_num_pred_arg2->type_enum = PT_TYPE_INTEGER;
+  ins_num_pred_arg2->info.value.data_value.i = 1;
+
+  ins_num_pred->info.expr.arg1 = ins_num_pred_arg1;
+  ins_num_pred->info.expr.arg2 = ins_num_pred_arg2;
+
+  parser_append_node (ins_num_pred, node->info.query.q.select.where);
```

Comment:
```suggestion
  node->info.query.q.select.where = parser_append_node (ins_num_pred, node->info.query.q.select.where);
```

WHERE 절이 NULL인 경우에는 추가가 되지 않아서 수정이 필요합니다.

```sql
drop table if exists t1;
create table t1 (c1 int);
insert into t1 values (1);

drop table if exists t2;
create table t2 (c1 int, c2 int);
insert into t2
with recursive cte (n) as (
    select 1
    union all
    select n + 1 from cte where n < 2000
  )
select 1, rownum from cte;

set trace on;
select * from t1 a where exists (select 1 from t2 b);

Trace Statistics:
  SELECT (time: 1, fetch: 7, fetch_time: 0, ioread: 0)
    SCAN (table: dba.t1), (heap time: 0, fetch: 2, ioread: 0, readrows: 1, rows: 1)
    SUBQUERY (uncorrelated)
      SELECT (time: 1, fetch: 5, fetch_time: 0, ioread: 0)
        SCAN (table: dba.t2), (heap time: 1, fetch: 5, ioread: 0, readrows: 2000, rows: 2000)
```

이 예시를 보면 WHERE 절이 없어서 `(inst_num()<= ?:0 )`가 추가되지 않았습니다.

Reply:
78fc4d2
에서 반영되었습니다.

Reply:
확인했습니다. 감사합니다.

Reply:
현재(5/22) commit (a0b8267)에서 inst_num()에 대한 조건이 2번 붙습니다.

Reply:
동작이 의도한대로 된다면 고치지 않아도 되지 않나 싶습니다.

Reply:
Resolve 하겠습니다. (참고:  https://github.com/CUBRID/cubrid/pull/6192/files#r2106466556)